projects/core/src/site-theme/providers/site-theme-initializer-providers.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { APP_INITIALIZER, Provider } from '@angular/core';
+import { SiteThemeInitializer } from '../services/site-theme-initializer';
+
+export function initializeTheme(
+  siteThemeInitializer: SiteThemeInitializer
+): () => void {
+  const result = () => {
+    siteThemeInitializer.initialize();
+  };
+  return result;
+}
```

Comment:
```suggestion
export function initializeTheme(): () => void {
  const siteThemeInitializer = inject(SiteThemeInitializer);
  return () => {
    siteThemeInitializer.initialize();
  };
}
```

then in the provider: 
```
{
  provide: APP_INITIALIZER,
  useFactory: initializeTheme,
  multi: true,
},
```
Such a change helps prevent breaking changes in `initializeTheme` function in case of new injection is needed.

Reply:
thanks